### PR TITLE
mbedtls@2: deprecate

### PR DESCRIPTION
**Pull request by SMillerDev**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/m/mbedtls@2.rb
Comment (chenrui333) [inline Formula/m/mbedtls@2.rb:1]: we can also remove the livecheck per this change

**Comment by chenrui333** [comment]
we can also remove the livecheck per this change

**Comment by SMillerDev** [comment]
There might still be releases though. It's deprecated, not abandoned.

**Review by chenrui333 — APPROVED** [review]

**Comment by SMillerDev** [comment]
That's a disappointing amount of dependencies.
```console
  dislocker
    * Dependency 'mbedtls@2' is deprecated but has un-deprecated dependents. Either
      un-deprecate 'mbedtls@2' or deprecate it and all of its dependents.
  hashlink
    * Dependency 'mbedtls@2' is deprecated but has un-deprecated dependents. Either
      un-deprecate 'mbedtls@2' or deprecate it and all of its dependents.
  haxe
    * Dependency 'mbedtls@2' is deprecated but has un-deprecated dependents. Either
      un-deprecate 'mbedtls@2' or deprecate it and all of its dependents.
  julia
    * Dependency 'mbedtls@2' is deprecated but has un-deprecated dependents. Either
      un-deprecate 'mbedtls@2' or deprecate it and all of its dependents.
  neko
    * Dependency 'mbedtls@2' is deprecated but has un-deprecated dependents. Either
      un-deprecate 'mbedtls@2' or deprecate it and all of its dependents.
  shadowsocks-libev
    * Dependency 'mbedtls@2' is deprecated but has un-deprecated dependents. Either
      un-deprecate 'mbedtls@2' or deprecate it and all of its dependents.
```

**Comment by SMillerDev** [comment]
Neko seems deprecated itself: https://github.com/HaxeFoundation/neko/issues/287

**Comment by SMillerDev** [comment]
Shadowsocks issues are piling up without any answers anywhere: https://github.com/shadowsocks/shadowsocks-libev

**Comment by SMillerDev** [comment]
Haxe has an issue open: https://github.com/HaxeFoundation/haxe/issues/11195
Julia has one open to replace the dependency: https://github.com/JuliaLang/julia/issues/48799

**Comment by cho-m** [comment]
> Dislocker seems abandoned: [Aorimn/dislocker#314](https://github.com/Aorimn/dislocker/issues/314)

They did try switching to `mbedtls` 3 but it broke Linux builds - revert in https://github.com/Aorimn/dislocker/commit/179c51be74510c79b2cbb0225ebcb5d3567a84bd

---

> Shadowsocks issues are piling up without any answers anywhere: https://github.com/shadowsocks/shadowsocks-libev

We can probably deprecate it when we feel like (probably at same time as `mbedtls@2` unless any activity). Upstream is mainly focusing on `shadowsocks-rust`. From summary:
* Bug-fix-only libev port of shadowsocks. Future development moved to shadowsocks-rust

---

> Neko seems deprecated itself: [HaxeFoundation/neko#287](https://github.com/HaxeFoundation/neko/issues/287)

The problem is `neko` is a dependency of `haxe`. Given our stance on deprecated SSL/TLS libs, we may just have to deprecate them all if they are not off of `mbedtls@2` by EOL.

---

> Julia has one open to replace the dependency: [JuliaLang/julia#48799](https://github.com/JuliaLang/julia/issues/48799)

Julia is the biggest issue. There isn't much progress on what they plan to do (e.g. switch to OpenSSL, update to MbedTLS 3, switch to BoringSSL, etc).

I assume they will decide on something before EOL given there are enough maintainers over there.

**Pull request closed without merging**